### PR TITLE
✨ Implement ProcessStarted Notification

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Send a generic message when a ProcessInstance starts
2. Send a message which contains the ProcessModelId of the started ProcessInstance

**Issues:**

Related: https://github.com/process-engine/process_engine_runtime/issues/134

PR: #207

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/runtime/engine/execute_process_service.ts [hunk 1]
```diff
@@ -253,6 +253,20 @@ export class ExecuteProcessService implements IExecuteProcessService {
 
     await this._saveCorrelation(identity, processInstanceConfig.correlationId, processInstanceConfig.processInstanceId, processModel.id);
 
+    /**
+     * TODO: Currently, the message which is send, when a Process ends is
+     * implement in the EndEvent Handlers.
```
Comment (ElRaptorus) [inline src/runtime/engine/execute_process_service.ts]: If you are referring to the general message about a finished ProcessInstance, I agree.
That message should be moved here.

Feel free to do so, if you wish. 😉

Comment (robpal94) [reply src/runtime/engine/execute_process_service.ts]: I would throw a more generic `ProcessEnded` Message here, where you can subscribe to, if you not interested an `EndEventReached` Message. But I actually think, that this should be implemented in a dedicated Pr 😄 .

File: src/runtime/engine/execute_process_service.ts [hunk 2]
```diff
@@ -253,6 +254,27 @@ export class ExecuteProcessService implements IExecuteProcessService {
 
     await this._saveCorrelation(identity, processInstanceConfig.correlationId, processInstanceConfig.processInstanceId, processModel.id);
 
+    const processStartedMessage: ProcessStartedMessage = new ProcessStartedMessage(processInstanceConfig.correlationId,
+      processModel.id,
+      processInstanceConfig.processInstanceId,
+      processInstanceConfig.startEvent.id,
+      null, processInstanceConfig.processToken);
```
Comment (cmg-dev) [inline src/runtime/engine/execute_process_service.ts]:
```suggestion
      null,
      processInstanceConfig.processToken);
```

What does `null` stand for?!

Comment (cmg-dev) [reply src/runtime/engine/execute_process_service.ts]: you could give it a name above it, like:

```ts

const xyz = null;

(...)
xyz,
processInstanceConfig.processToken);
(...)

Comment (robpal94) [reply src/runtime/engine/execute_process_service.ts]: The `flowNodeInstanceId` is not known when the MessageObject is created. Thats why I set this parameter to `null`.

Comment (ElRaptorus) [reply src/runtime/engine/execute_process_service.ts]: See my suggestion here:
https://github.com/process-engine/process_engine_contracts/pull/84#discussion_r235693289

Comment (ElRaptorus) [inline src/runtime/engine/execute_process_service.ts]: See my suggestion here:
https://github.com/process-engine/process_engine_contracts/pull/84#discussion_r235693289

Comment (SteffenKn) [reply src/runtime/engine/execute_process_service.ts]: I would still change it to something like
```
    const flowNodeInstanceId: string = null;

    const processStartedMessage: ProcessStartedMessage = new ProcessStartedMessage(processInstanceConfig.correlationId,
      processModel.id,
      processInstanceConfig.processInstanceId,
      processInstanceConfig.startEvent.id,
      flowNodeInstanceId,
      processInstanceConfig.processToken);
```

Comment (robpal94) [reply src/runtime/engine/execute_process_service.ts]: @SteffenKn see https://github.com/process-engine/process_engine_contracts/pull/84#discussion_r235693289 
This shouldn't be an issue anymore.

File: package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "@process-engine/external_task_api_contracts": "0.5.0",
```
Comment (ElRaptorus) [inline package.json]: Don't forget to update the `consumer_api_contracts` version here before merging.

**Pull request merged**